### PR TITLE
duc 1.4.3 (new formula)

### PR DESCRIPTION
**Pull request by oandrew**
Add formula for duc - a collection of tools for inspecting disk usage (http://duc.zevv.nl)

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/duc.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  url "https://github.com/zevv/duc/releases/download/1.4.3/duc-1.4.3.tar.gz"
+  sha256 "504810a1ac1939fb1a70bd25e492f91ea38bcd58ae0a962ce5d35559d7775e74"
+  head "https://github.com/zevv/duc.git"
+
+  option "with-opengl", "Enable OpenGL gui"
+
+  depends_on "pkg-config" => :build
+  depends_on "tokyo-cabinet"
+  depends_on "glfw" if build.with? "opengl"
+  depends_on "cairo" => :recommended
```
Comment (fxcoudert) [inline Formula/duc.rb]: Let's try to minimise the number of options, making `cairo` and `pango` non optional.

File: Formula/duc.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class Duc < Formula
+  desc "Suite of tools for inspecting disk usage"
+  homepage "http://duc.zevv.nl"
+  url "https://github.com/zevv/duc/releases/download/1.4.3/duc-1.4.3.tar.gz"
+  sha256 "504810a1ac1939fb1a70bd25e492f91ea38bcd58ae0a962ce5d35559d7775e74"
+  head "https://github.com/zevv/duc.git"
+
+  option "with-opengl", "Enable OpenGL gui"
```
Comment (fxcoudert) [inline Formula/duc.rb]: We don't want to provide 2 different GUIs, so either OpenGL is compatible with cairo/pango and we enable it all the time, or it's incompatible and we drop it.

File: Formula/duc.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          *opts
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/duc", "--version"
```
Comment (fxcoudert) [inline Formula/duc.rb]: Can we do something other than print version number? Like, output a graph to a file?

**Review by fxcoudert — DISMISSED** [review]

**Comment by oandrew** [comment]
@ fxcoudert I've made the changes you requested

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by oandrew** [comment]
@BrewTestBot test this please

**Comment by oandrew** [comment]
@BrewTestBot test this please

**Comment by oandrew** [comment]
@BrewTestBot test this please

**Comment by oandrew** [comment]
@ilovezfs @fxcoudert

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @oandrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**